hooks/useTreatments.ts

```diff
@@ -1,29 +1,29 @@
 import { Treatment } from '@/interfaces/treatment/Treatment';
 import { getTreatments } from '@/services/TreatmentService';
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 
-export const useTreatments = () => {
+export const useTreatments = (sort?: string) => {
     const [treatments, setTreatments] = useState<Treatment[]>([]);
     const [loading, setLoading] = useState(true);
     const [error, setError] = useState<string | null>(null);
 
-    useEffect(() => {
-        const fetchTreatments = async () => {
-            setLoading(true);
-            try {
-                const data = await getTreatments();
+    const fetchTreatments = useCallback(async () => {
+        setLoading(true);
+        try {
+            const data = await getTreatments(sort);
+            setTreatments(data);
+            setError(null);
```

Comment:
💅🏽   I'd add a conditional to only update the status if the error was previously set:

```suggestion
            if (error != null) { setError(null) }
```